Multiplatform/Navigation/Regular/SidebarView.swift

```diff
@@ -34,8 +36,14 @@ struct SidebarView: View {
             }
         } detail: {
             if let selection = selection {
-                NavigationStack {
+                NavigationStack(path: $navigationPath) {
```

Comment:
This is not required and should be handled by `selection`

Reply:
I don't think so. Handing this with `selection` will lose all current items on the stack and the back button will back to the top menu instead of the previous page the user is navigated to. This is not an expected behavior of navigation to me to lose everything on the stack just to see an album/artist.

Reply:
The problem is that it should open using the correct provider to match the iOS app and to make sure this works offline. And I don't think this is too bad as you are most likely to trigger this from the now playing screen, so you are not actively interacting with the current stack and have probably already reached your goal (starting playback).

Reply:
I come up with an alternate approach. I added a providerOverride option to Album related views. When this option is specified, the page will load with the specified data provider instead of the environment provided one. This will allow us to keep the navigation stack and load an offline albumView. Already tested with offline library and it works when the internet is down.

Reply:
I still don't particularly like this solution, can you give an example for an interaction where it would be desirable to keep the navigation stack?

Reply:
What could work is to just update the environment key here and don't use override parameters. I think it would be a good solution.

Reply:
> I still don't particularly like this solution, can you give an example for an interaction where it would be desirable to keep the navigation stack?

When adding songs to a playlist, I want my navigation stack to be retained because I don't want to search for a certain album/track again. Navigating using the now playing view is an infrequent but valid use case because the song currently playing's particular album may contain another song I want to add, and when I go back I want to see what I perviously viewed.

> What could work is to just update the environment key here and don't use override parameters. I think it would be a good solution.

This will cause navigation error because when you go back to an album that is not downloaded it will error out as the environment value is changed.

Reply:
After more intensive testing, I've decided to revert the idea of keeping the stack for now and just use selection instead. The current approach is fragile and could break in some cases, potentially causing a weird stack. It's better to have no stack than to have a broken one, so I'll use selection until I come up with a proper navigation solution. This isn't a trivial issue and shouldn't be addressed as a fix but rather as a new feature. For a fix for no navigation, the selection approach is good enough.

Reply:
While testing this I found some odd behavior in the iOS app as, well: when you navigate using notifications the current view gets replaced rather than the new view getting appended to the stack. Its kind of weird because the new view is added to it using `append`